runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -8674,12 +8674,12 @@ class INTERPRETER_CLASS
 		 * a field ref as resolved if only one of the two values has been set (by
 		 * another thread that is in the middle of a resolve).
 		 */
-		if (J9_UNEXPECTED((flags <= valueOffset) || J9_ARE_NO_BITS_SET(flags, J9FieldFlagPutResolved))) {
```

Comment:
why is the `J9FieldFlagPutResolved` check removed

Reply:
ah I see, its only set with the `J9_RESOLVE_FLAG_FIELD_SETTER` flag

Reply:
Minor: we can start using `!VM_VMHelpers::instanceFieldRefIsResolved(flags, valueOffset)` for this check

Reply:
Updated.